src/api/procedures/modifyMultiSig.ts

```diff
@@ -0,0 +1,154 @@
+import BigNumber from 'bignumber.js';
+
+import { PolymeshError, Procedure } from '~/internal';
+import { ErrorCode, MultiSig, Signer, TxTags } from '~/types';
+import { ProcedureAuthorization } from '~/types/internal';
+import { signerToSignatory, signerToString, stringToAccountId } from '~/utils/conversion';
+import { checkTxType } from '~/utils/internal';
+
+/**
+ * @hidden
+ */
+export interface ModifyMultiSigParams {
+  /**
+   * The MultiSig to be modified
+   */
+  multiSig: MultiSig;
+  /**
+   * The signers to set for the MultiSig
+   */
+  signers: Signer[];
+}
+
+export interface Storage {
+  signersToAdd: Signer[];
+  signersToRemove: Signer[];
+  requiredSignatures: BigNumber;
+}
+
+/**
+ * @hidden
+ */
+function calculateSignerDelta(
+  current: Signer[],
+  target: Signer[]
+): Pick<Storage, 'signersToAdd' | 'signersToRemove'> {
+  const currentSet = new Set(current.map(signerToString));
+  const targetSet = new Set(target.map(signerToString));
+
+  const newSigners = new Set([...target].filter(s => !currentSet.has(signerToString(s))));
+  const removedSigners = new Set([...current].filter(s => !targetSet.has(signerToString(s))));
+
+  return { signersToAdd: Array.from(newSigners), signersToRemove: Array.from(removedSigners) };
+}
+
+/**
+ * @hidden
+ */
+export async function prepareModifyMultiSig(
+  this: Procedure<ModifyMultiSigParams, void, Storage>,
+  args: ModifyMultiSigParams
+): Promise<void> {
+  const {
+    context: {
+      polymeshApi: { tx },
+    },
+    storage: { signersToAdd, signersToRemove, requiredSignatures },
+    context,
+  } = this;
+  const { signers, multiSig: multi } = args;
+
+  const [signingIdentity, creator] = await Promise.all([
+    context.getSigningIdentity(),
+    multi.getCreator(),
+  ]);
+
+  if (signersToAdd.length === 0 && signersToRemove.length === 0) {
```

Comment:
```suggestion
  if (!signersToAdd.length && !signersToRemove.length) {
```
(nit) Can leave as it is as well